src/main/java/org/kiwiproject/dropwizard/poller/ClientPoller.java

```diff
@@ -0,0 +1,492 @@
+import static org.kiwiproject.concurrent.Async.withMaxTimeout;
+import static org.kiwiproject.jaxrs.KiwiResponses.closeQuietly;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.setup.Environment;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+import org.kiwiproject.dropwizard.poller.metrics.ClientPollerStatistics;
+import org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatistics;
+
+import javax.ws.rs.client.SyncInvoker;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Utilizes the core client to poll a specified endpoint
+ */
+@Builder
+@Slf4j
+public class ClientPoller {
+
+    /**
+     * The default timeout value when polling synchronously
+     */
+    public static final long DEFAULT_SYNC_RESPONSE_CONSUMER_TIMEOUT = 5;
+
+    /**
+     * The default timeout unit when polling synchronously
+     */
+    public static final TimeUnit DEFAULT_SYNC_RESPONSE_CONSUMER_TIMEOUT_UNIT = TimeUnit.MINUTES;
+
+    /**
+     * The default timeout value for the {@link #supplier}
+     */
+    public static final long DEFAULT_SUPPLIER_TIMEOUT = 90;
+
+    /**
+     * The default timeout unit for the {@link #supplier}
+     */
+    public static final TimeUnit DEFAULT_SUPPLIER_TIMEOUT_UNIT = TimeUnit.SECONDS;
+
+    /**
+     * The default duration to wait before the first execution
+     */
+    public static final Duration DEFAULT_INITIAL_EXECUTION_DELAY = Duration.ofSeconds(5);
+
+    /**
+     * Whether to poll using fixed rate or with fixed delay, as defined by {@link ScheduledExecutorService}.
+     * <p>
+     *
+     * @implNote This enum should only have TWO values corresponding to the two ways of scheduling on a
+     * {@link ScheduledExecutorService}. There is no way to enforce that, however.
+     * @see ScheduledExecutorService#scheduleAtFixedRate(Runnable, long, long, TimeUnit)
+     * @see ScheduledExecutorService#scheduleWithFixedDelay(Runnable, long, long, TimeUnit)
+     */
+    public enum DelayType {
+        FIXED_RATE, FIXED_DELAY
+    }
+
+    /**
+     * The URI the poller is polling. <em>Not required, but recommended</em>
+     * <p>
+     * NOTE: only used for logging purposes. It will be overwritten so we should not ever manually set it.
```

Comment:
I thought you were going to do something about this hypocrisy 😜 (i.e. recommended but it gets overwritten)